### ISSUE TITLE
Weird online eval metric and loss performance during training?

### ISSUE THREAD

**Issue by RuntimeErrorz** [ShuweiShao/NDDepth#18]
![image](https://github.com/ShuweiShao/NDDepth/assets/71929239/e8dfd222-6fb6-4b97-8403-71cad863a95d)
From the image above, it seems that abs rel is rather low even at the beginning, and in the end of training it doesn't get any lower. The loss also didn't converage well.
By the way, is this final evaluation result normal？
![image2](https://github.com/ShuweiShao/NDDepth/assets/71929239/ff7f2774-9567-4b08-acd2-8c05fd5000c9)

**Comment by ShuweiShao** [comment]
HI, it seems strange.  You can check the txt file in the same directory as the saved model to obtain the online eval results. Can you reproduce the results by directly loading the given model weights?

**Comment by RuntimeErrorz** [comment]
Thank you again for you instant reply.
The text file shows the same result as the tensorboard, and the existing model weights can reproduce the same result in paper.

**Comment by ShuweiShao** [comment]
The results are abnormal. This is my online eval log during training:
[1130_logs.txt](https://github.com/ShuweiShao/NDDepth/files/14065589/1130_logs.txt)
[1201_logs.txt](https://github.com/ShuweiShao/NDDepth/files/14065590/1201_logs.txt)

Which training set did you use?

**Comment by RuntimeErrorz** [comment]
arguments_train_nyu.txt

**Comment by ShuweiShao** [comment]
Have you changed something in the code?

**Comment by RuntimeErrorz** [comment]
Nothing is changed.

**Comment by ShuweiShao** [comment]
Did you load the pretrained weights on ImageNet?

**Comment by RuntimeErrorz** [comment]
Yes, as the `arguments_train_nyu.txt` says `--pretrain model_zoo/swin_transformer/swin_large_patch4_window7_224_22k.pth
`

**Comment by ShuweiShao** [comment]
That's so strange.

**Comment by ShuweiShao** [comment]
If you have any process, please tell me . Thank you!

**Comment by RuntimeErrorz** [comment]
It turns out that small `batch_size` could affect the result heavily. I didn't realize that until I switched into a new server. I used to believe that it may take a longer time for a small `batch_size`  model to converge, but it won't affect the final result. Thank you for your help and it is truly a good project.

**Comment by ShuweiShao** [comment]
Glad to hear that!
